### PR TITLE
[dashboard] Add tooltip to project names

### LINKED ISSUE #5122: [dashboard] Long project names are cut of which makes them hard to tell apart
When opening a long repo name, e.g. `github.com/some-long-org/business-unit-and-department` the context on the dashboard is cut of and it's very hard to tell workspaces apart.

### PR DESCRIPTION
**Pull request by laushinka**
Fixes #5122

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
Fixes #5122

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx
Comment (gtsiolis) [inline components/dashboard/src/workspaces/WorkspaceEntry.tsx:94]: **issue**: Could we enforce a max width for the tooltip component as the tooltip overflows the container width when the path is long?

Frame 38

**Review by gtsiolis — APPROVED** [review]
Looks good @laushinka! 🏀 

Left one minor comment regarding UX. Feel free to open a follow up issue for this if you think it will significantly increase the scope of this PR. 🏓

Cross-posting https://github.com/gitpod-io/gitpod/issues/5122#issuecomment-903692584 for visibility. 👀

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 80ffa61b199b45e6f24eb1a822819b4c4bbb1203</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5323#pullrequestreview-736214247" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
@gtsiolis At first I set a max width for it, but then I decided against it since it's a temporary solution and we wanted users to be able to see the whole link for now. Let me know what you think :)

File: components/dashboard/src/workspaces/WorkspaceEntry.tsx
Comment (laushinka) [reply components/dashboard/src/workspaces/WorkspaceEntry.tsx:94]: Haha I wrote my comment just when you wrote this one.

**Comment by gtsiolis** [comment]
> ... but then I decided against it since it's a temporary solution and we wanted users to be able to see the whole link for now.

Sounds mergeable! I'd only add a max width and allow text wrapping to avoid overflowing the body container. Feel free to pass this to someone for taking a closer look at the code, too.

**Comment by gtsiolis** [comment]
**note**: Because of https://github.com/gitpod-io/gitpod/pull/5323#issuecomment-903844023, you'll also need an approval from someone from the engineering team. Cc @laushinka

**Comment by laushinka** [comment]
> > ... but then I decided against it since it's a temporary solution and we wanted users to be able to see the whole link for now.
> 
> @laushinka Sounds mergeable! I'd only add a max width and allow text wrapping to avoid overflowing the body container. Feel free to pass this to someone for taking a closer look at the code, too. Skipping the max width could also suffice for now. Let's see what other think.

No worries, definitely no strong feelings against adding the max-width and wrapping it. Will make the changes now :)

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

File: components/dashboard/src/components/Tooltip.tsx
Comment (JanKoehnlein) [inline components/dashboard/src/components/Tooltip.tsx:12]: Can we rename this to `allowWrap`  or similar?

Comment (laushinka) [reply components/dashboard/src/components/Tooltip.tsx:12]: Good suggestion.

**Comment by JanKoehnlein** [comment]
Thanks! Could you squash the changes?

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: b51b65f5e959cae631b685631373f576eac837c3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5323#pullrequestreview-736214247" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5323#issuecomment-904492381" title="LGTM">JanKoehnlein</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5323#pullrequestreview-736214247" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5323#issuecomment-904492381" title="LGTM">JanKoehnlein</a>*

Associated issue: *#5122*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 2caa4efb0e156bc3cd335b6883318951455ce7ee</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5323#pullrequestreview-736214247" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5323#issuecomment-905281433" title="LGTM">JanKoehnlein</a>*

Associated issue: *#5122*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
